### PR TITLE
Make the root of dylan-lang.github.io useful

### PR DESCRIPTION
**Pull request by cgay**
Adds an index to the various packages for which documentation is published via GitHub Actions.

### CODE REVIEW

**Comment by cgay** [comment]
We should automate this, too, so that we only have to edit `index.rst` without the manual step to generate `index.html`....later.

**Comment by fraya** [comment]
As I understand, you want a `index.rst` with a link to the `github.io` repositories that generate the documentation in GH actions, and when this index file is changed, an action generates a webpage and uploads it to `dylan-lang.github.io`, isn´t it?

I think I would add an sphinx project with an `index.rst` and a GH action that executes `linkcheck` (so we know that the link is correct), construct the document with the 'furo' theme and upload to `dylan-lang.github.io` the result. I would add opendylan's logo. The `index.html` will not be necessary. What do you think?

File: README.rst
Comment (fraya) [inline README.rst:1]: I think I would add an sphinx project with an `index.rst` and a GH action that executes `linkcheck` (so we know that the links are correct), construct the document with the 'furo' theme and upload to dylan-lang.github.io the result (same as the other projects). I would add opendylan's logo. The index.html will not be necessary. What do you think?

Comment (cgay) [reply README.rst:1]: That sounds great.

**Pull request merged**